### PR TITLE
Add a sample for FocusTraversalGroup

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

Added a new sample code block for `FocusTraversalGroup`.

## Related Issues

- https://github.com/flutter/flutter/issues/45076

## Tests

- No tests needed: this is a documentation change.

## Breaking Change

- [X] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/focus_traversal.dart [hunk 1]
```diff
@@ -1420,6 +1420,155 @@ class FocusTraversalOrder extends InheritedWidget {
...
+/// numerical order, the second follows a lexical order (ordered to traverse
+/// right to left), and the third ignores the numerical order assigned to it and
+/// traverses in widget order.
+///
+/// ```dart preamble
+/// /// A button wrapper that adds either a numerical or lexical order, depending on
+/// /// the type of T.
+/// class OrderedButton<T> extends StatefulWidget {
+///   const OrderedButton(
+///       {this.name,
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/focus_traversal.dart]: Nit: The formatting here looks a bit off.

File: packages/flutter/lib/src/widgets/focus_traversal.dart [hunk 2]
```diff
@@ -1420,6 +1420,155 @@ class FocusTraversalOrder extends InheritedWidget {
...
+///         children: <Widget>[
+///           // A group that is ordered with a numerical order, from left to right.
+///           FocusTraversalGroup(
+///             policy: OrderedTraversalPolicy(),
+///             child: Row(
+///               mainAxisAlignment: MainAxisAlignment.center,
+///               children: List<Widget>.generate(3, (int index) {
+///                 return OrderedButton<num>(
+///                   name: 'num: $index',
+///                   // TRY THIS: change this to "3 - index" and see how the order changes.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/focus_traversal.dart]: I really like the combination of interactive dartpad samples and the `TRY THIS` options. Really cool.

File: packages/flutter/lib/src/widgets/focus_traversal.dart [hunk 3]
```diff
@@ -1420,6 +1420,155 @@ class FocusTraversalOrder extends InheritedWidget {
...
+///                   name: 'String: $order',
+///                   order: order,
+///                 );
+///               }),
+///             ),
+///           ),
+///           // A group that orders in widget order, regardless of what the order is set to.
+///           FocusTraversalGroup(
+///             // Note that because this is NOT an OrderedTraversalPolicy, these
+///             // "OrderedButtons" their assigned order is ignored, and they are
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/focus_traversal.dart]:
```suggestion
///             // "OrderedButtons" assigned order is ignored, and they are
```
Nit: extra word, maybe? Not sure what is off about this sentence. You choose! :)

Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/focus_traversal.dart]: Thanks, reworded it.

**Review by Piinks — APPROVED** [review]
LGTM! 🎉

**Pull request merged**